### PR TITLE
BRIDGE 1226: WIP towards full onboarding support

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
This is a WIP pull request that adds to the onboarding. This adds a step for registration.

Remaining:
1. Share step (used when user is ineligible)
2. Lots of missing images
3. Email verification
4. Forgot password
5. Logo on "All Set" step

### CODE REVIEW

File: BridgeAppSDK/SBAConsentReviewStep.swift [hunk 1]
```diff
@@ -251,5 +255,29 @@ class SBAConsentReviewStepViewController: ORKPageStepViewController {
...
+        guard let consentResult = self.consentReviewResult where consentResult.isConsented else {
+            let error = NSError(domain: "SBAConsentReviewStepDomain",
+                                code: -1,
+                                userInfo: [NSLocalizedDescriptionKey : Localization.localizedString("SBA_REGISTRATION_NOT_CONSENTED")])
+            self.delegate?.stepViewControllerDidFail(self, withError: error)
+            return
+        }
+        
+        // set the consent to the shared user
+        sharedUser.consentSignature = consentResult.consentSignature
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBAConsentReviewStep.swift:277]: Some studies (e.g. the resilience project) are going to want to have the user sign multiple separate consent forms. Not saying we have to support this today, just food for thought.

Comment (syoung-smallwisdom) [reply BridgeAppSDK/SBAConsentReviewStep.swift:277]: A valid point.  However, that makes it much harder to remain compatible with AppCore (which I currently need to do b/c mPower uses the `APCUser` object). Until all functionality of Onboarding and user Profile is supported by BridgeAppSDK, this consent flow can only be used to handle default consent.

File: BridgeAppSDK/SBARegistrationStep.swift [hunk 2]
```diff
@@ -37,24 +37,27 @@ public class SBARegistrationStep: ORKFormStep, SBAProfileInfoForm {
     
     static let confirmationIdentifier = "confirmation"
     
+    static let defaultPasswordMinLength = 4
+    static let defaultPasswordMaxLength = 16
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBARegistrationStep.swift:41]: not seeing where these are used

**Comment by Erin-Mounts** [comment]
couple of questions/nits, otherwise looks good, merge when ready

File: BridgeAppSDK/SBAProfileInfoForm.swift [hunk 3]
```diff
@@ -145,38 +148,48 @@ public struct SBAProfileInfoOptions {
...
                 // confirmation
                 if (surveyItemType == .account(.registration)) {
-                    let confirmIdentifier = "\(formItem.identifier).\(SBARegistrationStep.confirmationIdentifier)"
-                    let confirmFormItem = passwordFormItem.confirmationAnswerFormItemWithIdentifier(confirmIdentifier,
-                                                                                                    text: Localization.localizedString("CONFIRM_PASSWORD_FORM_ITEM_TITLE"),
-                                                                                                    errorMessage: Localization.localizedString("CONFIRM_PASSWORD_ERROR_MESSAGE"))
+                    
+                    // If this is a registration, go ahead and set the default password verification
+                    let minLength = SBARegistrationStep.defaultPasswordMinLength
+                    let maxLength = SBARegistrationStep.defaultPasswordMaxLength
```
Comment (syoung-smallwisdom) [inline BridgeAppSDK/SBAProfileInfoForm.swift:180]: default min and max passwords are used in the protocol which can't have static variables defined on it. Therefore, I put them on the registration step.

**Pull request merged**